### PR TITLE
Add whoami command

### PR DESCRIPTION
**Pull request by ersen0**

### CODE REVIEW

File: bin/whoami [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+Author: Oğuz Ersen, oguzersen@protonmail.com
+License: perl
+
+=end metadata
+
+=cut
+
+use strict;
+
+my $user = getpwuid($>);
```
Comment (briandfoy) [inline bin/whoami:18]: Note that getpwuid doesn't work on Windows, so you'll need another way to handle that for that platform.

File: bin/whoami [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+=head1 SEE ALSO
+
+whoami(1), id(1)
+
+=head1 COPYRIGHT and LICENSE
+
+Copyright (c) by Oğuz Ersen 2021.
+
+This program is free and open software. You may use, modify, distribute,
+and sell this program (and any modified variants) in any way you wish,
```
Comment (briandfoy) [inline bin/whoami:47]: Note that the declared license in the metadata is "perl", so there are some restrictions there.

**Review by briandfoy — CHANGES_REQUESTED** [review]

File: bin/whoami (hunk 1 above)
Comment (briandfoy) [reply bin/whoami:18]: Or, maybe it's more complicated:

https://stackoverflow.com/questions/3526420/how-do-i-get-the-current-user-in-perl-in-a-portable-way

Comment (ersen0) [reply bin/whoami:18]: I am considering:

    my $user = $^O eq 'MSWin32' ? $ENV{USERNAME} : getpwuid($>);

I am not sure if it will work since I don't have access to a Windows machine right now.

File: bin/whoami (hunk 2 above)
Comment (ersen0) [reply bin/whoami:47]: I have copied and pasted this from other commands, e.g. `bin/awk` has the same. Is it OK if I change this section with (also copied and pasted from other commands, i.e. `bin/fold`):

    This code is freely modifiable and freely redistributable under Perl's
    Artistic License.

File: bin/whoami (hunk 1 above)
Comment (briandfoy) [reply bin/whoami:18]: Don't settle on anything until you know it will work on Windows. Note the strategies that I noted and what's mentioned in that Stackoverflow post.

File: bin/whoami (hunk 2 above)
Comment (briandfoy) [reply bin/whoami:47]: It would be better to simply say "This code is licensed under the Artistic License 2."

File: bin/whoami (hunk 1 above)
Comment (ersen0) [reply bin/whoami:18]: How about:

```
my $user;

if ($^O eq 'MSWin32') {
    $user = getlogin() || $ENV{USERNAME};
}
else {
    $user = getpwuid($>);
}

print "$user\n";
```

Or should I use `Win32::LoginName`?:

```
my $user;

if ($^O eq 'MSWin32') {
    require Win32;
    $user = Win32::LoginName;
}
else {
    $user = getpwuid($>);
}

print "$user\n";
```

Comment (briandfoy) [reply bin/whoami]: I don't know the answers to these. You'll need to try these out in various situations on Windows to see if they work.

But, remember that one of the goals is that we only use core modules. Optionally using a non-core module is fine if the operation doesn't depend on that.

File: bin/whoami [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+=cut
+
+use strict;
+
+my $user;
+
+if ($^O eq 'MSWin32') {
+    require Win32;
+    $user = Win32::LoginName();
```
Comment (briandfoy) [inline bin/whoami:22]: handle the case where Win32::* is not available. See the `getlogin()` that I showed you earlier. Remember, you are mostly limited to core modules.

**Review by briandfoy — CHANGES_REQUESTED** [review]

File: bin/whoami (hunk 3 above)
Comment (ersen0) [reply bin/whoami]: I changed this part with `$user = getlogin() || $ENV{USERNAME};`, Is this enough or are there edge cases where both `getlogin()` returns empty and `USERNAME` environment variable is not available? I can test it as soon as I have access to a Windows machine but it may take a while.

Comment (briandfoy) [reply bin/whoami]: Once you get a chance to work with this on Windows, see what works.

But, if you don't have access to Windows, is there some other reason you'd like this program to be in PerlPowerTools? Is there another non-unix system you are targeting that doesn't already have a `whoami`?

Comment (ersen0) [reply bin/whoami]: I just saw it was missing and wanted to contribute (: This can be closed without merging if it is not suitable for PerlPowerTools. Thanks for your guidance and patience.

**Comment by briandfoy** [comment]
No, it's a totally appropriate addition. We just have to figure out how to get it right on Windows. Major pain in the butt (see #89 for example).

**Comment by briandfoy** [comment]
I've merged this manually and did a little work on it. It will be part of the next release. Thanks,
